### PR TITLE
Parsing operations v 0.1.0

### PR DESCRIPTION
**Pull request by kct949**
this is some very rough draft of the code that can be used for translating user input into operations. i tried to cover more ground rather than focus on doing things the right way, so there is a lot of hacks and shortcuts at the moment. 

OperationInputTests -> FullFlowTest() showcases the current state of the implementation. It assumes user input as follows:

```
{
          "id": "operation-id",
          "type": "http",
          "httpMethod": "post",
          "headers": [
            {
             "key": "user-id",
             "value": "000001" 
            }
          ],
          "queryParameters": [
            {
             "key": "queryparam1",
             "value": "abcdef" 
            }
          ],
          "arguments": [
            {
                "name": "arg1",
                "required": true,
                "type": "number"
            },
            {
                "name": "arg2",
                "required": true,
                "type": "string"
            },
            {
                "name": "arg3",
                "required": true,
                "type": "null"
            },
            {
                "name": "arg4",
                "required": true,
                "type": "array"
            },
            {
                "name": "this-will-get-replaced-with-random-string",
                "required": false,
                "type": "string"
            }
          ],
          "payload": {
            "type": "json",
            "content": {
              "key1": {
                "key2": "{{arg1}}",
                "key3": [ "{{this-will-get-replaced-with-random-string}}", "{{arg2}}" ],
                "key4": "{{arg3}}",
                "key5": "{{arg4}}",
                "key6": "static-value"
              }
            }
          },
          "url": "https://httpbin.org/anything/{{arg2}}/@@key1.key6@@"
        }
```

and a dictionary of arguments that would normally be passed down from transactions or scenarios:

``` 
{
            { "arg1", 1 },
            { "arg2", "2" },
            { "arg3", null },
            { "arg4", new[] {5, 6, 7} },
            // notice no 'this-will-get-replaced-with-random-string' here - since it was not required, we will just autogenerate some value instead 
        }
```

This results in (this is a response from httpbin which basically mirrors whatever was sent to it):
image

so to sum up:
1. we support referencing arguments ({{arg}}) in payloads (and we support different types of arguments)
2. we support referencing arguments ({{arg}}) and payload fields (@@path.to.payload.property@@) in URLs
3. we support specifying payloads with properties that will get auto-generated by us 
4. we support passing headers and query params (these cannot access args/payloads etc at the moment)

all of these still need work, but it's a start

i also worked on the test setup a bit to make it easier for myself to test a lot of different inputs without having to repeat a lot of test code.

### CODE REVIEW

File: src/MicroservicesSimulationFramework/MicroservicesSimulationFramework.Core/Models/Input/TransactionOperationInput.cs [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+public enum ArgumentType
+{
+    String,
+    Number,
+    Object,
+    Array,
+    Boolean,
+    Null
+}
```
Comment (kct949) [inline src/MicroservicesSimulationFramework/MicroservicesSimulationFramework.Core/Models/Input/TransactionOperationInput.cs:85]: add values here
